### PR TITLE
use node-convict for configuration; reorg directory structure to bin/,lib/; use const/let as appropriate

### PR DESCRIPTION
**Pull request by jrgm**
- put libs in lib/, and executables in bin/
- no-var
- use node-convict to be consistent with other repos

The config changes are one step to moving this someday to standard dockerflow.

r? - @jbuck or @vladikoff

### CODE REVIEW

File: lib/config.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  maximumRcptTo: {
+    'default': 5,
+    doc: 'Maximum number of RCPT TO on one connection dialogue.',
+    env: 'MAXIMUM_RCPT_TO',
+    format: 'int'
+  },
+
+  // Domain whitelist. RCPT TO addresses must match one of the domains on this
+  // list. (Strict matching, no subdomains).
+  rcptToDomainWhitelist: {
```
Comment (vladikoff) [inline lib/config.js]: 👍

File: lib/config.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    'default': os.tmpdir(),
+    doc: 'Location of directory in which to store webmaster, etc. email.',
+    env: 'SPECIAL_USER_DIR',
+    format: String
+  },
+
+  webPort: {
+    'default': 8080,
+    doc: 'Port number for web server to listen on.',
+    env: 'WEB_PORT',
```
Comment (vladikoff) [inline lib/config.js]: nit: I like the `EMAIL_PORT`, but i am not sure if we have to specify the `WEB_` here, can we just use `PORT`?

Comment (jrgm) [reply lib/config.js]: We could. But symmetry? I find it odd that one gets to be PORT and the other not.

Comment (vladikoff) [reply lib/config.js]: symmetry++

**Pull request merged**